test/unit/test_env_variable.py

```diff
@@ -0,0 +1,171 @@
+
+
+def test_var_not_set_same_logging_file(set_tempdir):
+    """
+    Integration test which verifies that re-using the same logging for multiple tasks works as expected,
+    using the default log path
+    """
+    tasks = run_n_simple_tasks(5)
+    for task in tasks:
+        assert task["log_path"] == default_log_path(task["jobid"])
+
+
+def test_custom_log_path_points_at_file(set_tempdir):
+    """
+    Integration test which verifies that using a custom log path for a single task works if the path points to a file
+    """
+    temp_dir = set_tempdir
+    custom_log_path = Path(temp_dir) / "main.log"
+    os.environ[LOG_ENV_VARIABLE_NAME] = str(custom_log_path)
+    tasks = run_n_simple_tasks(1)
+    assert tasks[0]["log_path"] == custom_log_path
+    os.environ.pop(LOG_ENV_VARIABLE_NAME)
+
+
+def test_preexisting_custom_log_file(set_tempdir):
+    """
+    Integration test which verifies that using a custom log path for a single task works if file already exists.
+    """
+    temp_dir = set_tempdir
+    custom_log_path = Path(temp_dir) / "main.log"
+    file_content = "This existing file has content."
+    with open(custom_log_path, "a") as f:
+        f.write(file_content)
+    os.environ[LOG_ENV_VARIABLE_NAME] = str(custom_log_path)
+    tasks = run_n_simple_tasks(1)
+    assert tasks[0]["log_path"] == custom_log_path
+    with open(custom_log_path, "r") as f:
+        content = f.read()
+        assert file_content in content
+    os.environ.pop(LOG_ENV_VARIABLE_NAME)
+
+
+def test_same_logging_file_custom_log_path(set_tempdir):
+    """
+    Integration test which verifies that re-using the same logging for multiple tasks works as expected,
+    using a custom log path
+    """
+    temp_dir = set_tempdir
+    custom_log_path = Path(temp_dir) / "main.log"
+    os.environ[LOG_ENV_VARIABLE_NAME] = str(custom_log_path)
+    tasks = run_n_simple_tasks(5)
+    for task in tasks:
+        assert task["log_path"] == custom_log_path
+    os.environ.pop(LOG_ENV_VARIABLE_NAME)
+
+
+def test_different_custom_logging_file(set_tempdir):
+    """
+    Integration test which verifies that changing the log path from one invocation of run_task to the next will work
+    """
+    temp_dir = set_tempdir
+    task_creator = lambda: generate_root_task(task_class=TestTask, x="Test")
+
+    use_specific_log_file(task_creator, temp_dir, "first")
+    use_specific_log_file(task_creator, temp_dir, "second")
+
+
+def test_custom_log_path_points_at_dir(set_tempdir):
+    """
+    Integration test which verifies that using a custom log path for a single task fails if the path points to a directory instead of a file
+    """
+    temp_dir = set_tempdir
+    custom_log_path = Path(temp_dir)
+    os.environ[LOG_ENV_VARIABLE_NAME] = str(custom_log_path)
+
+    with pytest.raises(Exception) as e:
+        run_n_simple_tasks(1)
+
+    assert e.type.__name__ == "IsADirectoryError"
+    os.environ.pop(LOG_ENV_VARIABLE_NAME)
```

Comment:
```suggestion
```

not needed with the mock_settings_env_vars fixture and also save, if the test fails